File: Modul 6/C/PRAK604-2310817110007-MuhammadDaffaMusyafa.c
#include <stdio.h>
#include <string.h>

int main() {
    char pesan1[100];
    char pesan2[100];
    int i;
    int j;
    int asterik = 0;
    int hash = 0;

    scanf(" %99[^\n]", pesan1);
    scanf(" %99[^\n]", pesan2);

    size_t panjangpesan1 = strlen(pesan1);
    size_t panjangpesan2 = strlen(pesan2);

    i = 0;
    j = 0;

    while (i < panjangpesan1 && j < panjangpesan2) {
        if (pesan1[i] == ' ' && pesan2[j] == ' ') {
            i++;
            j++;
            continue;
        }
        if (pesan1[i] == pesan2[j]) {
            asterik++;
        } else {
            hash++;
        }

        i++;
        j++;
    }

    if (panjangpesan1 == panjangpesan2) {
        printf("\n* = %d\n", asterik);
        printf("# = %d\n", hash);
        if (asterik >= hash) {
            printf("Pesan Asli");
        } else {
            printf("Pesan Palsu");
        }
    } else {
        printf("Panjang kalimat berbeda, pesan palsu");
    }
    return 0;
}

File: Modul 6/py/PRAK604-2310817110007-MuhammadDaffaMusyafa.py
pesan1 = input()
pesan2 = input()

i = 0
j = 0
asterik = 0
hash = 0

while i < len(pesan1) and j < len(pesan2):
    if pesan1[i] == ' ' and pesan2[j] == ' ':
        i += 1
        j += 1
        continue
    if pesan1[i] == pesan2[j]:
        asterik += 1
    else:
        hash += 1

    i += 1
    j += 1

panjang_pesan1 = len(pesan1)
panjang_pesan2 = len(pesan2)

if panjang_pesan1 == panjang_pesan2:
    print("\n* =", asterik)
    print("# =", hash)
    if asterik >= hash:
        print("Pesan Asli")
    else:
        print("Pesan Palsu")
else:
    print("Panjang kalimat berbeda, pesan palsu")
